source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -569,9 +569,12 @@ bool Filter::maybeTunnel(Upstream::ThreadLocalCluster& cluster) {
             config_->regexEngine()),
         std::unique_ptr<Http::NullRouteImpl>);
   }
-  generic_conn_pool_ = factory->createGenericConnPool(
-      cluster, config_->tunnelingConfigHelper(), this, *upstream_callbacks_,
-      upstream_decoder_filter_callbacks_, getStreamInfo());
+  auto host = cluster.chooseHost(this);
+  if (host) {
+    generic_conn_pool_ = factory->createGenericConnPool(
+        host, cluster, config_->tunnelingConfigHelper(), this, *upstream_callbacks_,
+        upstream_decoder_filter_callbacks_, getStreamInfo());
+  }
```

Comment:
In the case where hose is null, should we reset `generic_conn_pool_`? I'm looking at the old code and it seems to unconditionally assign to  `generic_conn_pool_`

Reply:
yeah I was torn on this.  on the one hand I think it doesn't make sense to try to create a generic connection pool wrapping a known nullptr host as it's destined to fail.  on the other hand it would make it a slightly lower risk refactor.  have a preference?

Reply:
Not a strong one, but since this isn't runtime guarded, I would probably be inclined to err on the side of lower risk. But I do agree that it's likely to fail either way.

Reply:
Ah so  I took another look and what would happen and what would happen is the conn pool created would fail validity checks in createGenericConnPool and we'd return nullptr, so this is equivalen and avoids some work